exchange/wrappers/buzzex.js

```diff
@@ -0,0 +1,335 @@
+];
+
+// errors that might mean
+// the API call succeeded.
+const unknownResultErrors = [
+  'Response code 502',
+  'Response code 504',
+  'Response code 522',
+  'Response code 520',
+]
+
+const includes = (str, list) => {
+  if(!_.isString(str))
+    return false;
+
+  return _.some(list, item => str.includes(item));
+}
+
+Trader.prototype.handleResponse = function(funcName, callback, nonMutating, payload) {
+  return (error, body) => {
+
+    if(!error && !body) {
+      error = new Error('Empty response');
+    }
+
+    if(error) {
+      if(includes(error.message, recoverableErrors)) {
+        error.notFatal = true;
+      }
+
+      if(includes(error.message, ['Rate limit exceeded'])) {
+        error.notFatal = true;
+        error.backoffDelay = 2500;
+      }
+
+      if(nonMutating && includes(error.message, unknownResultErrors)) {
+        // this call only tried to retrieve data, safe to redo
+        error.notFatal = true;
+      }
+
+      //condition here...
+
+      return callback(error);
+    }
+    
+    return callback(undefined, body);
+  }
+};
+
+Trader.prototype.getTrades = function(since, callback, descending) {
+  const startTs = since ? moment(since).valueOf() : null;
+
+  const handle = (err, trades) => {
+
+    if (err) return callback(err);
+
+    var parsedTrades = [];
+    _.each(trades[this.pair], function(trade) {
+      // Even when you supply 'since' you can still get more trades than you asked for, it needs to be filtered
+      //if (_.isNull(startTs) || startTs < moment.unix(trade[4]).valueOf()) {
+        parsedTrades.push({
+          tid: trade.tid,
+          date: parseInt(Math.round(trade.timestamp), 10),
+          price: parseFloat(trade.price),
+          amount: parseFloat(trade.amount)
+        });
+      //}
+    }, this);
+
+    if(descending)
+      callback(undefined, parsedTrades.reverse());
+    else
+      callback(undefined, parsedTrades);
+  };
+
+  const reqData = {
+    param: this.pair
+  };
+
+  console.log(since);
+
+  if(since) {
+    reqData.param = this.pair + "/" +(since/1000);
+  }
+
+  const fetch = cb => this.buzzex.api('trades', reqData, this.handleResponse('getTrades', cb, true));
+  
+  retry(null, fetch, handle);
+};
+
+Trader.prototype.getPortfolio = function(callback) {
+  const handle = (err, data) => {
+    if(err) return callback(err);
+
+
+    var portfolio = [];
+    _.each(data.funds, function(currency) {
+      
+      portfolio.push({ name: currency[0], amount: currency[1] });
+      
+    }, this);
+
+    return callback(undefined, portfolio);
+  };
+
+  const fetch = cb => this.buzzex.api('getinfo', {}, this.handleResponse('getPortfolio', cb, true));
+  retry(null, fetch, handle);
+};
+
+// This assumes that only limit orders are being placed with standard assets pairs
+// It does not take into account volume discounts.
+// Base maker fee is 0.16%, taker fee is 0.26%.
+Trader.prototype.getFee = function(callback) {
+  const makerFee = 0.16;
```

Comment:
These are the kraken fees, not the Buzzex fees according to: https://support.buzzex.io/hc/en-us/articles/360008709594-What-Is-The-Fee-Structure-On-Buzzex-